### PR TITLE
Update Calloop to 0.10.0

### PR DESCRIPTION
**Pull request by wash2**
This PR updates Calloop to 0.10.0

### CODE REVIEW

File: src/event_loop.rs [hunk 1]
```diff
@@ -125,23 +121,23 @@ impl EventSource for WaylandSource {
         &mut self,
         poll: &mut calloop::Poll,
         token_factory: &mut TokenFactory,
-    ) -> std::io::Result<()> {
+    ) -> Result<(), calloop::Error> {
```
Comment (i509VCB) [inline src/event_loop.rs:124]: This should be io::Result?

File: src/event_loop.rs [hunk 2]
```diff
@@ -171,7 +167,7 @@ impl EventSource for WaylandSource {
         Ok(())
     }
 
-    fn post_run<F>(&mut self, _: F) -> std::io::Result<()>
+    fn post_run<F>(&mut self, _: F) -> Result<(), calloop::Error>
```
Comment (i509VCB) [inline src/event_loop.rs:170]: calloop::Result<()> can be used here

File: src/event_loop.rs (hunk 1 above)
Comment (wash2) [reply src/event_loop.rs]: when I try to use `io::Result`, I see an error. `calloop::Result` is fine though

**Comment by vberger** [comment]
Given this is a breaking change of the API, I'd be inclined to integrating it as part of the 0.30 rework, rather than make a breaking calloop release for just that.

**Pull request closed without merging**